File: app/src/main/java/leon/homework/LoginActivity.java
package leon.homework;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;
import android.widget.Toast;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class LoginActivity extends AppCompatActivity implements View.OnClickListener{
    private ImageButton imgbtn_back;
    private Button btn_login;
    private Button btn_register;
    private Button btn_forgetpwd;
    private EditText et_phone;
    private EditText et_pwd;
    private TextView tv_massage_phone;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        imgbtn_back = (ImageButton)findViewById(R.id.imgbtn_back);
        btn_login = (Button)findViewById(R.id.btn_login);
        btn_register = (Button)findViewById(R.id.btn_register);
        btn_forgetpwd = (Button)findViewById(R.id.btn_forgetpwd);
        imgbtn_back.setOnClickListener(this);
        btn_login.setOnClickListener(this);
        btn_register.setOnClickListener(this);
        btn_forgetpwd.setOnClickListener(this);
        et_phone = (EditText)findViewById(R.id.et_phone);
        et_pwd = (EditText)findViewById(R.id.et_pwd);
        tv_massage_phone = (TextView)findViewById(R.id.tv_message_phone);
    }

    public void checkpwd(){
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    String data = null;
                    String params = "act=1";
                    params = params +"&vid="+2;
                    URL url = new URL("http://zz503379238.oicp.net/index.php");
                    HttpURLConnection conn = (HttpURLConnection)url.openConnection();
                    StringBuilder sb = new StringBuilder();
                    BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream()));
                    String msg = "";
                    while((msg = br.readLine())!=null){
                        sb.append(msg);
                    }
                    data = sb.toString();
                    Toast.makeText(null,data,Toast.LENGTH_SHORT).show();
                    Log.d(null, "run: >>>>>>>>>>"+data);
                } catch (IOException e) {
                    e.printStackTrace();
                }
                Message msg = new Message();
                msg.what=1;
                handler.sendMessage(msg);
            }
        }).start();
    }

    private Handler handler = new Handler(){
        public void handleMessage(Message msg){
            if (msg.what==1){
                Intent intent = new Intent(LoginActivity.this,MainActivity.class);
                startActivity(intent);
                finish();
            }
        }
    };

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.imgbtn_back:
                finish();
                break;
            case R.id.btn_login:
                checkpwd();
                break;
            case R.id.btn_register:
                Intent intent2 = new Intent(LoginActivity.this,RegisterActivity.class);
                startActivity(intent2);
                finish();
                break;
            case R.id.btn_forgetpwd:
                break;
        }
    }
}

File: app/src/main/java/leon/homework/Activities/LoginActivity.kt
package leon.homework.Activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_login.*
import leon.homework.AppContext
import leon.homework.Data.Const
import leon.homework.Data.SaveData
import leon.homework.Extensions.loading
import leon.homework.Person.Student
import leon.homework.R
import org.jetbrains.anko.async
import org.jetbrains.anko.toast
import org.jetbrains.anko.uiThread

class LoginActivity : BaseActivity(), View.OnClickListener {
    var stu_user: Student? = null
    override val layoutResourceId: Int = R.layout.activity_login

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        imgbtn_back.setOnClickListener(this)
        btn_login.setOnClickListener(this)
        btn_register.setOnClickListener(this)
        btn_forgetpwd.setOnClickListener(this)
        if(AppContext.stuUser!=null){
            stu_user = AppContext.stuUser!!
        }else{
            AppContext.stuUser = Student()
            stu_user = AppContext.stuUser!!
        }
    }

    override fun onClick(v: View) {
        val id = v.id
        when (id) {
            R.id.imgbtn_back -> finish()
            R.id.btn_login -> {
                if(et_phone.text.length==11&&et_pwd.text.length>=8){
                    loading(Const.SHOW)
                    CheckPwd()
                }else
                    toast("请输入正确格式手机号或密码")
            }
            R.id.btn_register -> {
                startActivity(Intent(LoginActivity@this, RegisterActivity::class.java))
                finish()
            }
            R.id.btn_forgetpwd -> {
                startActivity(Intent(LoginActivity@this, ForgetPwdActivity::class.java))
                finish()
            }
        }
    }

    fun CheckPwd(){
        async() {
            val result = stu_user!!.Login(et_phone.text.toString(),et_pwd.text.toString())
            uiThread {
                loading(Const.HIDE)
                when(result){
                    Const.LOGIN_SUCCESS->{
                        SaveData.isLogined = true
                        sendBroadcast(Intent("FinishWelcome"));
                        startActivity(Intent(this@LoginActivity, MainActivity::class.java))
                        finish()
                    }
                    Const.LOGIN_NEED_COMPLETE->{
                        startActivity(Intent(this@LoginActivity, CompleteInfoActivity::class.java))
                        finish()
                    }
                    Const.LOGIN_NOEXIST->{
                        toast("此手机号不存在")
                    }
                    Const.LOGIN_WRONGPWD->{
                        toast("密码错误")
                    }
                    Const.LOGIN_ERROR->{
                        toast("服务器比较辣鸡，连接服务器失败")
                    }
               }
            }
        }
    }
}
